mabe.go

```diff
@@ -0,0 +1,256 @@
+package cfdi
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/i18n"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/gobl/regimes/mx"
+)
+
+// Mabe schema constants
+const (
+	MabeVersion        = "1.0"
+	MabeNamespace      = "https://recepcionfe.mabempresa.com/cfd/addenda/v1"
+	MabeSchemaLocation = "https://recepcionfe.mabempresa.com/cfd/addenda/v1/mabev1.xsd"
+	MabeNotApplicable  = "NA"
+)
+
+// TipoDocumento valid values
+const (
+	MabeTipoDocumentoFactura     = "FACTURA"
+	MabeTipoDocumentoNotaCredito = "NOTA CREDITO"
+	MabeTipoDocumentoNotaCargo   = "NOTA CARGO"
+)
+
+// MabeFactura is the root element of the Mabe addendum
+type MabeFactura struct {
+	XMLName        xml.Name `xml:"mabe:Factura"`
+	Namespace      string   `xml:"xmlns:mabe,attr"`
+	SchemaLocation string   `xml:"xsi:schemaLocation,attr"`
+
+	Version       string `xml:"version,attr"`
+	TipoDocumento string `xml:"tipoDocumento,attr"`
+	Folio         string `xml:"folio,attr"`
+	Fecha         string `xml:"fecha,attr"`
+	OrdenCompra   string `xml:"ordenCompra,attr"`
+	Referencia1   string `xml:"referencia1,attr"`
+	Referencia2   string `xml:"referencia2,attr,omitempty"`
+
+	Moneda    *MabeMoneda     `xml:"mabe:Moneda"`
+	Proveedor *MabeProveedor  `xml:"mabe:Proveedor"`
+	Entrega   *MabeEntrega    `xml:"mabe:Entrega"`
+	Detalles  *[]*MabeDetalle `xml:"mabe:Detalles>mabe:Detalle"`
+
+	Descuentos  *MabeDescuentos  `xml:"mabe:Descuentos,omitempty"`
+	Subtotal    *MabeImporte     `xml:"mabe:Subtotal"`
+	Traslados   *[]*MabeImpuesto `xml:"mabe:Traslados>mabe:Traslado"`
+	Retenciones *[]*MabeImpuesto `xml:"mabe:Retenciones>mabe:Retencion"`
+	Total       *MabeImporte     `xml:"mabe:Total"`
+}
+
+// MabeMoneda carries the data about the invoice's currency
+type MabeMoneda struct {
+	TipoMoneda      string `xml:"tipoMoneda,attr"`
+	TipoCambio      string `xml:"tipoCambio,attr,omitempty"`      // Not implemented yet
```

Comment:
I left this one out for the moment, as we need to add support for multiple currencies to the CFDI itself first and I didn't want to delay things any further. Happy to implement it right after if we think it's worth the extra time now.